File: RepoTransBench/target_projects/Java/Go/code4craft_hello-design-pattern/tests/original/structural/proxy/hello_world_proxy_test.go
package proxy

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type HelloWorld interface {
	HelloWorld() string
}

type DefaultHelloWorld struct{}

func (d *DefaultHelloWorld) HelloWorld() string {
	return "Hello Proxy!"
}

type HelloWorldProxy struct {
	target HelloWorld
}

func NewHelloWorldProxy(target HelloWorld) *HelloWorldProxy {
	return &HelloWorldProxy{target: target}
}

func (h *HelloWorldProxy) HelloWorld() string {
	return h.target.HelloWorld()
}

func TestHelloWorldFacade(t *testing.T) {
	proxy := NewHelloWorldProxy(&DefaultHelloWorld{})
	assert.Equal(t, "Hello Proxy!", proxy.HelloWorld())
}

File: RepoTransBench/target_projects/Java/C#/code4craft_hello-design-pattern/tests/original/structural/proxy/HelloWorldProxyTest.cs
using Xunit;

namespace HelloDesignPattern.Tests.structural.proxy
{
    public class HelloWorldProxyTest
    {
        [Fact]
        public void TestHelloWorldFacade()
        {
            var helloWorldProxy = new HelloWorldProxy(new HelloWorldProxy.DefaultHelloWorld());
            Assert.Equal("Hello Proxy!", helloWorldProxy.HelloWorld());
        }
    }
}